benchflow-dsl/src/main/scala/cloud/benchflow/dsl/definition/configuration/goal/GoalYamlProtocol.scala

```diff
@@ -18,9 +18,12 @@ import scala.util.Try
 object GoalYamlProtocol extends DefaultYamlProtocol {
 
   val TypeKey = YamlString("type")
+  val StoredKnowledgeKey = YamlString("stored_knowledge")
   val ObservationKey = YamlString("observation")
   val ExplorationSpaceKey = YamlString("exploration_space")
 
+  val StoredKnowledgeDefaultValue = true
```

Comment:
Is this the final way by which we are going to handle defaults? I would have an object holding default values, so that they are clearly identifiable in the class.